flow360/component/simulation/translator/solver_translator.py

```diff
@@ -0,0 +1,664 @@
+        "ForchheimerCoefficient": list(model_dict["forchheimerCoefficient"]),
+    }
+    if model.volumetric_heat_source is not None:
+        porous_medium["volumetricHeatSource"] = model_dict["volumetricHeatSource"]
+
+    return porous_medium
+
+
+# pylint: disable=too-many-branches
+def boundary_spec_translator(model: SurfaceModelTypes, op_acousitc_to_static_pressure_ratio):
+    """Boundary translator"""
+    model_dict = remove_units_in_dict(dump_dict(model))
+    boundary = {}
+    if isinstance(model, Wall):
+        boundary.update(
+            {
+                "type": "WallFunction" if model.use_wall_function else "NoSlipWall",
+                "velocityType": model.velocity_type,
+            }
+        )
+        if model.velocity:
+            boundary["velocity"] = model_dict["velocity"]
+        if isinstance(model.heat_spec, Temperature):
+            boundary["temperature"] = model_dict["heatSpec"]["value"]
+        elif isinstance(model.heat_spec, HeatFlux):
+            boundary["heatFlux"] = model_dict["heatSpec"]["value"]
+    elif isinstance(model, Inflow):
+        boundary["totalTemperatureRatio"] = model_dict["totalTemperature"]
+        if model.velocity_direction:
+            boundary["velocityDirection"] = model_dict["velocityDirection"]
+        if isinstance(model.spec, TotalPressure):
+            boundary["type"] = "SubsonicInflow"
+            boundary["totalPressureRatio"] = (
+                model_dict["spec"]["value"] * op_acousitc_to_static_pressure_ratio
+            )
+        elif isinstance(model.spec, MassFlowRate):
+            boundary["type"] = "MassInflow"
+            boundary["massFlowRate"] = model_dict["spec"]["value"]
+    elif isinstance(model, Outflow):
+        if isinstance(model.spec, Pressure):
+            boundary["type"] = "SubsonicOutflowPressure"
+            boundary["staticPressureRatio"] = (
+                model_dict["spec"]["value"] * op_acousitc_to_static_pressure_ratio
+            )
+        elif isinstance(model.spec, Mach):
+            pass
+        elif isinstance(model.spec, MassFlowRate):
+            pass
+    elif isinstance(model, Periodic):
+        pass
+    elif isinstance(model, SlipWall):
+        boundary["type"] = "SlipWall"
+    elif isinstance(model, Freestream):
+        boundary.update(
+            {
+                "type": "Freestream",
+                "velocityType": model.velocity_type,
+            }
+        )
+
+    return boundary
+
+
+# pylint: disable=too-many-statements
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-locals
+@preprocess_input
+def get_solver_json(
+    input_params: SimulationParams,
+    # pylint: disable=no-member
+    mesh_unit: LengthType.Positive,
+):
+    """
+    Get the solver json from the simulation parameters.
+    """
+
+    translated = {}
+    ##:: Step 1: Get geometry:
+    geometry = remove_units_in_dict(dump_dict(input_params.reference_geometry))
```

Comment:
@benflexcompute 
reference_geometry is optional and default is None, it will break here